src/components/JobManagementTable.tsx

```diff
@@ -0,0 +1,205 @@
+import {
+  Table,
+  TableBody,
+  TableCell,
+  TableHead,
+  TableHeader,
+  TableRow,
+} from './ui/table';
+import { DeleteIcon, Edit } from 'lucide-react';
+import { getAllJobsAdditonalType, JobType } from '@/types/jobs.types';
+import { ServerActionReturnType } from '@/types/api.types';
+import { useToast } from './ui/use-toast';
+import {
+  Dialog,
+  DialogContent,
+  DialogDescription,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+  DialogTrigger,
+} from '@/components/ui/dialog';
+import { Button } from './ui/button';
+import { JobQuerySchemaType } from '@/lib/validators/jobs.validator';
+import { DEFAULT_PAGE, JOBS_PER_PAGE } from '@/config/app.config';
+import { Pagination, PaginationContent, PaginationItem } from './ui/pagination';
+import {
+  PaginationNextButton,
+  PaginationPreviousButton,
+} from './pagination-client';
+import APP_PATHS from '@/config/path.config';
+import { PaginationPages } from './ui/paginator';
+import {
+  approveJob,
+  deleteJobById,
+  getAllJobsForAdmin,
+} from '@/actions/job.action';
+import ApproveJobDialog from './ApproveJobDialog';
+
+type props = {
+  searchParams: JobQuerySchemaType;
+  jobs: ServerActionReturnType<getAllJobsAdditonalType>;
+  initialJobs: JobType[];
+};
+
+const JobManagementTable = ({ jobs, searchParams, initialJobs }: props) => {
+  const { toast } = useToast();
+  const [DispJobs, setDispJobs] = useState<JobType[]>(initialJobs);
+
+  const handleDelete = async (JobId: string) => {
+    try {
+      const result = await deleteJobById({ id: JobId });
+      if (result.status) {
+        setDispJobs((prevJobs) => prevJobs.filter((job) => job.id !== JobId));
+        toast({ title: result.message });
+      }
+    } catch (error: any) {
+      toast({
+        title: 'Failed to delete job',
+        description: error.message,
+        variant: 'destructive',
+      });
+    }
+  };
+
+  const handleApproveJob = async (jobId: string) => {
+    try {
+      const result = await approveJob({ id: jobId });
+      if (result.status) {
+        const item = DispJobs.find((job) => job.id === jobId);
+        if (item) {
+          const updatedJob = DispJobs.map((job) =>
+            job.id === jobId ? { ...job, isVerifiedJob: true } : job
+          );
+          setDispJobs(updatedJob);
+        }
+      }
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  const refreshJobs = async () => {
+    const refreshedJobs = await getAllJobsForAdmin(searchParams);
+    if (refreshedJobs.status && Array.isArray(refreshedJobs.additional?.jobs)) {
+      setDispJobs(refreshedJobs.additional?.jobs);
+    }
+  };
+
+  useEffect(() => {
+    refreshJobs();
+  }, [searchParams]);
+
+  if (!jobs.status) {
+    return <div>Error {jobs.message}</div>;
+  }
+
+  const totalPages =
+    Math.ceil((jobs.additional?.totalJobs || 0) / JOBS_PER_PAGE) ||
+    DEFAULT_PAGE;
+  const currentPage = parseInt(searchParams.page?.toString()) || DEFAULT_PAGE;
```

Comment:
why do we need the toString() here  , if you parse the search params correctly before passing it to this table component , we wont need it